File: FrontEnd/src/app/DTO/EncabezadoPlanillaDTO.ts
import { DetallePlanillaDTO } from "./DetallePlanillaDTO";

export class EncabezadoPlanillaDTO {
    encabezadoPlanillaId:number;
    estadoPlanillaId:number;
    periodoId:number;
    descripcion?: string;
    periodoDescripcion?:string;
    nombreEstado?:string;
    descripcionPeriodo?:string;
    aprobado?:boolean;
    enviarCorreo?:boolean;
    correoEnviado?:boolean;
    habilitado?:boolean;
    activo?: boolean;
    creado?: Date;
    detallePlanilla?: DetallePlanillaDTO[];

}

File: BackEnd/Planilla/DTO/Planilla/EncabezadoPlanillaDTO.cs
using Planilla.Entities;
using System.Collections.Generic;
using System;

namespace Planilla.DTO.Planilla
{
    public class EncabezadoPlanillaDTO
    {
        public int EncabezadoPlanillaId { get; set; }
        public int? PeriodoId { get; set; }
        public int? EstadoPlanillaId { get; set; }
        public string Descripcion { get; set; }
        public string NombreEstado { get; set; }
        public string DescripcionPeriodo { get; set; }
        public bool? Habilitado { get; set; }

        public DateTime? FechaCorte { get; set; }
        public bool? EnviarCorreo { get; set; }
        public bool? CorreoEnviado { get; set; }
        public bool? Activo { get; set; }
        public DateTime Creado { get; set; }
        public virtual IList<DetallePlanillaDTO> DetallePlanilla { get; set; } = new List<DetallePlanillaDTO>();

    }
}
